constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/motion/utils/ViewTimeCycle.java

```diff
@@ -13,58 +13,30 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package androidx.constraintlayout.motion.widget;
+package androidx.constraintlayout.motion.utils;
 
 import android.os.Build;
-
-import androidx.constraintlayout.widget.ConstraintAttribute;
-import androidx.constraintlayout.motion.utils.CurveFit;
-
 import android.util.Log;
 import android.util.SparseArray;
 import android.view.View;
 
+import androidx.constraintlayout.core.motion.utils.CurveFit;
+import androidx.constraintlayout.core.motion.utils.KeyCache;
+import androidx.constraintlayout.core.motion.utils.TimeCycleSplineSet;
+import androidx.constraintlayout.motion.widget.Key;
+import androidx.constraintlayout.motion.widget.MotionLayout;
+import androidx.constraintlayout.widget.ConstraintAttribute;
+
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
-import java.text.DecimalFormat;
-
-import androidx.constraintlayout.motion.utils.Oscillator;
 
 /**
  * This engine allows manipulation of attributes by wave shapes oscillating in time
  *
  * @hide
  */
-public abstract class TimeCycleSplineSet {
-    private static final String TAG = "SplineSet";
-    protected CurveFit mCurveFit;
-    protected int mWaveShape = 0;
-    protected int[] mTimePoints = new int[10];
-    protected float[][] mValues = new float[10][3];
-    private int count;
-    private String mType;
-    private float[] mCache = new float[3];
-    private static final int CURVE_VALUE = 0;
-    private static final int CURVE_PERIOD = 1;
-    private static final int CURVE_OFFSET = 2;
-    private static float VAL_2PI = (float) (2 * Math.PI);
-    protected boolean mContinue = false;
-    long last_time;
-    float last_cycle = Float.NaN;
-
-    @Override
-    public String toString() {
-        String str = mType;
-        DecimalFormat df = new DecimalFormat("##.##");
-        for (int i = 0; i < count; i++) {
-            str += "[" + mTimePoints[i] + " , " + df.format(mValues[i]) + "] ";
-        }
-        return str;
-    }
-
-    public void setType(String type) {
-        mType = type;
-    }
+public abstract class ViewTimeCycle extends TimeCycleSplineSet {
+    private static final String TAG = "ViewTimeCycle";
```

Comment:
It doesn't look like `TAG` is used
```suggestion
```

Reply:
Used in lines 232 and 234
 Log.e(TAG, "unable to setProgress", e);
Log.e(TAG, "unable to setProgress", e);